randovania/games/dread/logic_database/Cataris.txt

```diff
@@ -2814,7 +2814,12 @@ Extra - asset_id: collision_camera_040
   > Dock to Thermal Device Room North
       Can Slide
   > Event - Push Wide Beam Block
-      Wave Beam and Can Slide and Push Wide Beam Block
+      All of the following:
+          Wave Beam and Can Slide
+          Any of the following:
+              Push Wide Beam Block
+              # Diffusion Abuse: https://youtu.be/hJYbMcLUTX0
+              Wide Beam and Diffusion Abuse (Intermediate)
```

Comment:
Having this as diffusion abuse is all kinds of weird. We should introduce a new trick for using charge beam with a melee counter.

Reply:
The new trick would only be used here and for Artaria's. Is two uses enough to add a new trick? Different story if we find other uses for it.

Reply:
I asked before making the PR and the consensus was to keep it as Diffusion Abuse.

Reply:
To me it didn't look like that discussion ended with a consensus to file this under Diffusion Abuse. But if that's the route we're taking, can we at least put this at advanced so that it's above the pre existing usage?

Reply:
This is in fact not the exact same technique as the existing diffusion abuse trick. In the existing one, we use this setup to shoot an enemy and let the diffusion effect push a wide block. In this instance there is no diffusion involved, hence by reluctance to put this under diffusion abuse. Furthermore, the beginner diffusion abuse instances use the diffusion effect to trigger buttons, so the trick doesn't exclusively relate to wide beam blocks to begin with.

Up until now, the hypermode diffusion abuse trick was the only instance of using a melee counter for the purpose of shooting a charge beam without having a charge beam. If we introduce a trick for precisely that purpose, I think the old hypermode diffusion abuse trick should use both diffusion abuse and the new trick, which could be called something like "Charge from Counter". I'm not sure about how the difficulty load should be distributed between the two tricks, but I think it makes sense to take the diffusion abuse down from hypermode, while the primary difficulty level of each of these tricks can be placed on the charge from counter trick.

Also worth mentioning, by the logic that pushing wide beam blocks in exotic ways falls under diffusion abuse, the one next to the DNA hint station could also have been placed under that trick, but we have it under pseudo wave, even though it uses the wave beam.

I think it's worth it to add Charge from Counter as a new trick to accurately express the mechanisms at play instead of shoehorning different mechanics into the the same trick because they are used in tandem in some instances. If we discover more usages for using charge beam through a counter then I think this will look more clean.

Though, like I said, I can accept putting this as diffusion abuse advanced.

Reply:
I like Counter Charge better, everyone loves alliteration. I think it probably merits its own trick, but in my experience I've been discouraged to give tricks that only get one or two usage in the database. I bumped it to advanced for now.